File: app/src/main/java/grey/smarthouse/database/converters/IdConverter.java
package grey.smarthouse.database.converters;

import androidx.room.TypeConverter;

import java.util.UUID;

public class IdConverter {

    @TypeConverter
    public String idToString(UUID id) {
        if (id == null) {
            return null;
        } else {
            return id.toString();
        }
    }
    @TypeConverter
    public UUID stringToId(String string){
        return UUID.fromString(string);
    }

}

File: app/src/main/java/grey/smarthouse/database/converters/IdConverter.kt
package grey.smarthouse.database.converters

import androidx.room.TypeConverter
import java.util.*

class IdConverter {

    @TypeConverter
    fun idToString(id: UUID?): String? {
        return id?.toString()
    }

    @TypeConverter
    fun stringToId(string: String): UUID {
        return UUID.fromString(string)
    }

}
